File: src/main/cpp/plazma/kernel/lib/data/xml/Xml2JsonConverter.cpp
#include "Xml2JsonConverter.h"
#include "plazma/kernel/lib/data/json/jsonparser.h"
#include "plazma/kernel/lib/data/xml/xmlparser.h"
#include "plazma/kernel/lib/str/strlib.h"

namespace xml {

    Xml2JsonConverter::Xml2JsonConverter() {}

    Xml2JsonConverter::~Xml2JsonConverter() {}

    ////

    std::string Xml2JsonConverter::toString() {
        return "Xml2JsonConverter";
    }

    ////

    node::Node* Xml2JsonConverter::createDocumentNode() {
        return createObjectNode();
    }

    node::Node* Xml2JsonConverter::createObjectNode() {
        node::Node* node = new node::Node();
        node->setType(jsonparser::OBJECT_TYPE);
        return node;
    }

    node::Node* Xml2JsonConverter::createArrayNode() {
        node::Node* node = new node::Node();
        node->setType(jsonparser::ARRAY_TYPE);
        return node;
    }

    node::Node* Xml2JsonConverter::createAttributeNode() {
        node::Node* node = new node::Node();
        node->setType(jsonparser::ATTRIBUTE_TYPE);
        return node;
    }

}

File: src/main/java/plazma/lib/data/xml/Xml2JsonConverter.java
/*
 * Copyright (C) 2012-2023 Oleh Hapon ohapon@users.sourceforge.net
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307, USA.
 * 
 * Oleh Hapon
 * Kyiv, UKRAINE
 * ohapon@users.sourceforge.net
 */

package plazma.lib.data.xml;

import plazma.lib.data.json.JsonParser;
import plazma.lib.data.node.Node;

public class Xml2JsonConverter extends Xml2BaseConverter {

    public Node createDocumentNode() {
        return createObjectNode();
    }

    public Node createObjectNode() {
        Node node = new Node();
        node.setType(JsonParser.OBJECT_TYPE);
        return node;
    }

    public Node createArrayNode() {
        Node node = new Node();
        node.setType(JsonParser.ARRAY_TYPE);
        return node;
    }

    public Node createAttributeNode() {
        Node node = new Node();
        node.setType(JsonParser.ATTRIBUTE_TYPE);
        return node;
    }

    public String toString() {
        return "Xml2JsonConverter";
    }

}
